tools/topology/topology2/include/components/google-rtc-aec.conf

```diff
@@ -0,0 +1,138 @@
+#
+# Common widget google-rtc-aec
+#
+# A generic google-rtc-aec widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.google-rtc-aec.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.google-rtc-aec."N" {
+# 		index		1
+# 		type		"effect"
+#	}
+#
+# Where N is the unique instance number for the google-rtc-aec object within the same alsaconf node.
+Define {
+	AEC_UUID	"a6:a0:80:b7:9f:26:6f:46:b4:77:23:df:a0:5a:f7:58"
+}
+
+Class.Widget."google-rtc-aec" {
+	#
+	# Pipeline ID for the google-rtc-aec widget object
+	#
+	DefineAttribute."index" {}
+
+	#
+	# google-rtc-aec object instance
+	#
+	DefineAttribute."instance" {}
+
+	#include common component definition
+	<include/components/widget-common.conf>
+
+	DefineAttribute."num_audio_formats" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	#
+	# google-rtc-aec widget UUID
+	#
+	DefineAttribute."uuid" {
+		type	"string"
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.uuid"
+	}
+
+	DefineAttribute."cpc" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	DefineAttribute."is_pages" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_comp.word"
+	}
+
+	# Attribute categories
+	attributes {
+		#
+		# The google-rtc-aec widget name would be constructed using the index and instance attributes.
+		# For ex: "google-rtc-aec.1.1" or "google-rtc-aec.10.2" etc.
+		#
+		!constructor [
+			"index"
+			"instance"
+		]
+
+		#
+		# mandatory attributes that must be provided when the class is instantiated
+		#
+		!mandatory [
+			"no_pm"
+			"uuid"
+			"process_type"
+		]
+
+		#
+		# immutable attributes cannot be modified in the object instance
+		#
+		!immutable [
+			"uuid"
+			"type"
+		]
+
+		#
+		# deprecated attributes should not be added in the object instance
+		#
+		!deprecated [
+			"preload_count"
+		]
+
+		#
+		# google-rtc-aec widget objects instantiated within the same alsaconf node must have unique
+		# instance attribute
+		#
+		unique	"instance"
+	}
+
+	num_audio_formats 2
+	num_input_audio_formats 2
+	num_output_audio_formats 1
+	Object.Base.input_audio_format {
+		1 {
```

Comment:
any chance we give names or at least a comment so we know what endpoint is which?

Reply:
sure, added